xcodeproj/internal/xcschemes/xcschemes.bzl

```diff
@@ -0,0 +1,1125 @@
+            action as are built for the Run action (defined by
+            [`xcschemes.run`](#xcschemes.run)). If `None`, `xcschemes.profile()`
+            will be used, which means no targets will be built for the Profile
+            action.
+        run: A value returned by [`xcschemes.run`](#xcschemes.run).
+
+            If `None`, `xcschemes.run()` will be used, which means no targets
+            will be built for the Run action, except for `build_targets` and
+            `library_targets` specified in
+            [`xcschemes.profile`](#xcschemes.profile) and
+            [`xcschemes.test`](#xcschemes.test).
+        test: A value returned by [`xcschemes.test`](#xcschemes.test).
+
+            If `None`, `xcschemes.test()` will be used, which means no targets
+            will be built for the Test action.
+    """
+    if not name:
+        fail("""
+`name` must be provided to `xcschemes.scheme`.
+""")
+
+    return struct(
+        name = name,
+        profile = profile,
+        run = run,
+        test = test,
+    )
+
+# Actions
+
+def _profile(
+        *,
+        args = "inherit",
+        build_targets = [],
+        env = "inherit",
+        env_include_defaults = True,
+        launch_target = None,
+        use_run_args_and_env = None,
+        xcode_configuration = None):
+    """Defines the Profile action.
+
+    Args:
+        args: Command-line arguments to use when profiling the launch target.
+
+            If `"inherit"`, then the arguments will be supplied by the launch
+            target (e.g.
+            [`cc_binary.args`](https://bazel.build/reference/be/common-definitions#binary.args)).
+            Otherwise, the `list` of arguments will be set as provided, and
+            `None` or `[]` will result in no command-line arguments.
+
+            Each element of the `list` can either be a string or a value
+            returned by [`xcschemes.arg`](#xcschemes.arg). If an element is a
+            string, it will be transformed into `xcschemes.arg(element)`. For
+            example,
+            ```
+            xcschemes.profile(
+                args = [
+                    "-arg1",
+                    xcschemes.arg("-arg2", enabled = False),
+                ],
+            )
+            ```
+            will be transformed into:
+            ```
+            xcschemes.profile(
+                args = [
+                    xcschemes.arg("-arg1"),
+                    xcschemes.arg("-arg2", enabled = False),
+                ],
+            )
+            ```
+        build_targets: Additional targets to build when profiling.
+
+            Each element of the `list` can be a label string, a value returned
+            by
+            [`xcschemes.top_level_build_target`](#xcschemes.top_level_build_target),
+            or a value returned by
+            [`xcschemes.top_level_anchor_target`](#xcschemes.top_level_anchor_target).
```

Comment:
[nit]
```suggestion
            or [`xcschemes.top_level_anchor_target`](#xcschemes.top_level_anchor_target).
```

Reply:
Because of how the list is structured, it needs the same prefix. Think of it this way:

---

Each element of the `list` can be:

- a label string
- a value returned by `xcschemes.top_level_build_target`
- a value returned by `xcschemes.top_level_anchor_target`

---

Unless I'm just bad at grammar here?

Reply:
Nope. Fine with me.